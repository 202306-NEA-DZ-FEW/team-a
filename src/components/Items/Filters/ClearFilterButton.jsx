import React from "react";
import { MdFilterListOff } from "react-icons/md";

function ClearFilterButton({ t, handleClearFilters }) {
  return (
    <button
      onClick={handleClearFilters}
      className='btn btn-sm btn-primary bg-opacity-40 rounded-full normal-case font-light tracking-wider'
    >
      <MdFilterListOff />
      {t("itemsPage:clearFilter")}
    </button>
  );
}

export default ClearFilterButton;
