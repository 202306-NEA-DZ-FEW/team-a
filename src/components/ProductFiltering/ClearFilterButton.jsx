import { useRouter } from "next/router";
import React from "react";
import { MdFilterListOff } from "react-icons/md";

function ClearFilterButton({ t }) {
  const router = useRouter();
  return (
    <button
      onClick={() => {
        router.push("/products", undefined, { scroll: false });
      }}
      className='btn btn-sm rounded-full btn-neutral normal-case font-light tracking-wider'
    >
      <MdFilterListOff />
      {t("productsPage:clearFilter")}
    </button>
  );
}

export default ClearFilterButton;
