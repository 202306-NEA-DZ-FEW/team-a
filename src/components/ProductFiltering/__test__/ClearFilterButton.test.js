import renderer from "react-test-renderer";

import ClearFilterButton from "../ClearFilterButton";

it("renders correctly", () => {
  const tree = renderer.create(<ClearFilterButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
