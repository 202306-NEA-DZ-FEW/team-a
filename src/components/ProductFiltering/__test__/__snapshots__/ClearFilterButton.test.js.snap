// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn w-full md:w-40 bg-black text-white btn-sm px-4 rounded-xl"
  onClick={[Function]}
>
  Clear filter
</button>
`;
